File: backend/app/models/heatmap.py
from typing import Optional
from pydantic import BaseModel


class HeatmapStock(BaseModel):
    symbol: str
    name: str
    sector: str
    industry: str
    weight: float
    price: Optional[float] = None
    change_pct: Optional[float] = None


class HeatmapSector(BaseModel):
    name: str
    total_weight: float
    stocks: list[HeatmapStock]


class HeatmapData(BaseModel):
    index_symbol: str
    timestamp: str
    sectors: list[HeatmapSector]

File: frontend/src/types/heatmap.ts
export interface HeatmapStock {
  symbol: string
  name: string
  sector: string
  industry: string
  weight: number
  price: number | null
  change_pct: number | null
}

export interface HeatmapSector {
  name: string
  total_weight: number
  stocks: HeatmapStock[]
}

export interface HeatmapData {
  index_symbol: string
  timestamp: string
  sectors: HeatmapSector[]
}
